test/anim/evaluator/anim-cache.test.mjs

```diff
@@ -0,0 +1,80 @@
+import { AnimCache } from '../../../src/anim/evaluator/anim-cache.js';
+import { AnimData } from '../../../src/anim/evaluator/anim-data.js';
+import { INTERPOLATION_STEP, INTERPOLATION_LINEAR } from '../../../src/anim/constants.js';
+import { expect } from 'chai';
+
+describe('AnimCache', function () {
+    const animCache = new AnimCache();
+    const input = new AnimData(1, [0, 1, 2]);
+    const output = new AnimData(3, [0, 0, 0, 1, 2, 3, 2, 4, 6]);
+
+    describe('#constructor', function () {
+
+        it('instantiates correctly', function () {
+            expect(animCache).to.be.ok;
+        });
+
+    });
+
+    describe('#update', function () {
+
+        it('can update the normalized time of the cache', function () {
+            animCache.update(1.25, input.data);
+            expect(animCache._t).to.be.equal(0.25);
+        });
+
+    });
+
+    describe('#eval', function () {
+
+        it('can retrieve the step output keyframe value for a given input key', function () {
+            const result = [0, 0, 0];
+            animCache.update(0, input.data);
+            animCache.eval(result, INTERPOLATION_STEP, output);
+            expect(result[0]).to.be.equal(0);
+            expect(result[1]).to.be.equal(0);
+            expect(result[2]).to.be.equal(0);
```

Comment:
In order to be more succinct, you could do:

```
expect(result).to.deep.equal([0, 0, 0]);
```

Reply:
Quite a lot of these in your PR. If you agree the above is better, make sure you catch all of them. 😄